python/feathub/processors/flink/table_builder/sliding_window_utils.py

```diff
@@ -80,30 +65,67 @@ def to_java_descriptor(self) -> Any:
             gateway.jvm.java.time.Duration.ofMillis(
                 int(self.step_size.total_seconds() * 1e3)
             ),
-            self.limit,
             self.group_by_keys,
-            self.filter_expr,
         )
 
     def __eq__(self, other: object) -> bool:
         return (
             isinstance(other, self.__class__)
             and self.step_size == other.step_size
-            and self.limit == other.limit
             and self.group_by_keys == other.group_by_keys
-            and self.filter_expr == other.filter_expr
         )
 
     def __hash__(self) -> int:
-        return hash(
-            (
-                self.step_size,
-                self.limit,
-                tuple(self.group_by_keys),
-                self.filter_expr,
+        return hash((self.step_size, tuple(self.group_by_keys)))
+
+
+def evaluate_infinite_size_sliding_window(
+    flink_table: NativeFlinkTable,
+    window_descriptor: SlidingWindowDescriptor,
+    agg_descriptors: List[AggregationFieldDescriptor],
+    config: SlidingFeatureViewConfig,
+) -> NativeFlinkTable:
+    """
+    Evaluate the infinite size sliding window transforms on the given flink table and
+    return the result table.
+
+    :param flink_table: The input Flink table.
+    :param window_descriptor: The descriptor of the sliding window.
+    :param agg_descriptors: A list of descriptor that descriptor the aggregation to
+                            perform.
+    :param config: The config of the SlidingFeatureView that the window_descriptor
+                   belongs to.
+    :return: The result table.
+    """
+    for agg_descriptor in agg_descriptors:
+        if agg_descriptor.window_size != timedelta(seconds=0):
+            raise FeathubException(
+                "Sliding window with 0 step size must has 0 window size."
+            )
+        if not config.get(SKIP_SAME_WINDOW_OUTPUT_CONFIG):
+            raise FeathubException(
+                "When step size is 0, skip_same_window_output must be true."
+            )
```

Comment:
It seems a little weird to check this for every agg descriptor. Let's move this out of the for loop.